src/js/plugin.js

```diff
@@ -1,82 +1,44 @@
-function MediumEditorTable(options) {
-  this.options = extend(options, {
-    columns: 10,
-    rows: 10
-  });
-  this.parent = true;
-  this.hasForm = true;
-  this.isFormVisible = false;
-  this.createButton();
-}
+var MediumEditorTable = MediumEditor.extensions.form.extend({
+  name: 'table',
 
-MediumEditorTable.prototype = {
-  createButton: function () {
-    this._createButtonElement();
-    this._bindButtonClick();
-  },
-
-  isDisplayed: function () {
-    return this.isFormVisible;
-  },
-
-  getForm: function () {
-    if (!this.builder) {
-      this.builder = new Builder({
-        onClick: function (rows, columns) {
-          this.table.insert(rows, columns);
-          this.hideForm();
-        }.bind(this),
-        ownerDocument: this.document,
-        rows: this.options.rows,
-        columns: this.options.columns
-      });
-      this.table = new Table(this.base);
-    }
+  aria: 'create table',
+  action: 'table',
+  contentDefault: 'TBL',
+  contentFA: '<i class="fa fa-table"></i>',
 
-    return this.builder.getElement();
+  handleClick: function () {
+    this[this.isActive() === true ? 'hide' : 'show']();
   },
 
-  getButton: function () {
-    if (this.options.buttonLabel) {
-      this.button.innerHTML = this.options.buttonLabel;
-    } else if (this.base.options.buttonLabels === 'fontawesome') {
-      this.button.innerHTML = '<i class="fa fa-table"></i>';
-    }
-    return this.button;
-  },
-
-  onHide: function () {
-    this.hideForm();
-  },
-
-  hideForm: function () {
-    this.isFormVisible = false;
+  hide: function () {
+    this.setInactive();
     this.builder.hide();
-    this.button.classList.remove('medium-editor-button-active');
   },
 
   show: function () {
-    this.isFormVisible = true;
+    this.setActive();
     this.builder.show(this.button.offsetLeft);
-    this.button.classList.add('medium-editor-button-active');
     var elements = document.getElementsByClassName('medium-editor-table-builder-grid');
```

Comment:
You can use `this.document` here to get a reference to the `ownerDocument` from the editor.